### PR TITLE
Fix injection: prefer jaeger in the same namespace

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves https://issues.redhat.com/browse/TRACING-3722 

## Which problem is this PR solving?
- 

With this patch the agent injection code prefers Jaeger in the same namespace. 

Before if jaeger production (with a separate jaeger-query pod that injects agent) was created in multiple namespaces the jaeger-agent injected into the query pod was overridden to the Jaeger that was return as first from the API server. 

docker image: `docker.io/pavolloffay/jaeger-operator:agent-issue-PR2383-1`

## Description of the changes
See the comment in the code

## How was this change tested?
- 

## Checklist
- [ ] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [ ] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

**Comment by pavolloffay** [comment]
I will wait for @IshwarKanse confirmation on tests before merging

**Comment by IshwarKanse** [comment]
@pavolloffay Verified the bug fix and left a comment in the jira. We can proceed to merge this PR.

**Review by iblancasa — APPROVED** [review]

**Pull request merged**
